test_manager/test_wrapper/test_wrapper.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+
+	"cloud.google.com/go/storage"
+	junitFormatter "github.com/jstemmer/go-junit-report/formatter"
+	junitParser "github.com/jstemmer/go-junit-report/parser"
+)
+
+const (
+	metadataURLPrefix   = "http://metadata.google.internal/computeMetadata/v1/instance/attributes/"
+	testResultObject    = "outs/junit_go-test.xml"
+	testBinaryLocalPath = "image_test"
+	artifactPath        = "/artifact"
+	workDir             = "/workspace"
+)
+
+func main() {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("failed to create cloud storage client: %v", err)
+	}
+
+	testBinaryURL, err := getMetadataAttribute("_test_binary_url")
+	if err != nil {
+		log.Fatalf("failed to get metadata _test_binary_url: %v", err)
+	}
+
+	testRun, err := getMetadataAttribute("_test_run")
+
+	var testArguments = []string{"-test.v"}
+	if testRun != "" {
+		testArguments = append(testArguments, "-test.run", testRun)
+	}
+
+	if err = os.Mkdir(workDir+artifactPath, 0755); err != nil {
```

Comment:
don't you need to try creating workDir before creating workDir+artifactPath ?

Reply:
also lets have artifactPath be a full absolute path, so we don't have to remember how to build it later

Reply:
Good catch.